server/src/main/java/org/opensearch/search/backpressure/trackers/TaskResourceUsageTracker.java

```diff
@@ -41,10 +41,10 @@ public long getCancellations() {
     /**
      * Notifies the tracker to update its state when a task execution completes.
      */
-    public abstract void update(Task task);
+    public void update(Task task) {}
 
     /**
      * Returns the cancellation reason for the given task, if it's eligible for cancellation.
      */
-    public abstract Optional<TaskCancellation.Reason> cancellationReason(Task task);
+    public abstract Optional<TaskCancellation.Reason> checkAndMaybeGetCancellationReason(Task task);
```

Comment:
The function signature still doesn't look right, it doesn't clarify if cancellation will occur or not

Reply:
I still feel that it's correct – this only returns a `TaskCancellation.Reason` for a task *eligible* for cancellation. It doesn't say anything about whether the task will be *actually* cancelled or not.